src/internet_identity_interface/src/internet_identity/types/v2.rs

```diff
@@ -0,0 +1,65 @@
+use crate::internet_identity::types::{CredentialId, MetadataEntry, PublicKey, Timestamp};
+use candid::{CandidType, Deserialize};
+use std::collections::HashMap;
+
+pub type IdentityNumber = u64;
+
+#[derive(Clone, Debug, CandidType, Deserialize, Eq, PartialEq)]
+pub enum AuthnMethodProtection {
+    #[serde(rename = "protected")]
+    Protected,
+    #[serde(rename = "unprotected")]
+    Unprotected,
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize, Eq, PartialEq)]
+pub struct PublicKeyAuthn {
+    pub pubkey: PublicKey,
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize, Eq, PartialEq)]
+pub struct WebAuthn {
+    pub pubkey: PublicKey,
+    pub credential_id: CredentialId,
+}
+
+/// Supported authentication methods
+#[derive(Clone, Debug, CandidType, Deserialize, Eq, PartialEq)]
+pub enum AuthnMethod {
+    #[serde(rename = "webauthn")]
+    WebAuthn(WebAuthn),
+    #[serde(rename = "pubkey")]
+    PubKey(PublicKeyAuthn),
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize, Eq, PartialEq)]
+pub struct AuthnMethodData {
+    pub authn_method: AuthnMethod,
+    // contains now the following fields
+    // - alias
+    // - origin
+    // - key_type: reduced to "platform", "cross_platform" on migration
+    pub metadata: HashMap<String, MetadataEntry>,
+    pub protection: AuthnMethodProtection,
+    // last usage timestamp cannot be written and will always be ignored on write
+    pub last_authentication: Option<Timestamp>,
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize, Eq, PartialEq)]
+pub struct AuthnMethodRegistration {
+    pub expiration: Timestamp,
+    pub authn_method: Option<AuthnMethodData>,
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize, Eq, PartialEq)]
+pub struct IdentityInfo {
+    pub authn_methods: Vec<AuthnMethodData>,
+    pub recovery_authn_methods: Vec<AuthnMethodData>,
```

Comment:
Thanks a lot for the zoom-discussion.  As agreed, having the `AuthMethods` in a single list (with a `Purpose`-flag, as it is in api-v1) does have some advantages, so let's keep it this way.